packages/hooks/src/hooks/useTokenBalance.test.tsx

```diff
@@ -0,0 +1,161 @@
+import { renderHook } from '@testing-library/react-hooks';
+import React from 'react';
+import { useTokenBalance } from './useTokenBalance';
+import { Web3Context } from '../Provider';
+import { BigNumber, ethers } from 'ethers';
+
+jest.mock('ethers', () => ({
+  ...jest.requireActual('ethers'),
+  ethers: {
+    ...jest.requireActual('ethers').ethers,
+    Contract: class ContractStub {
+      balanceOf(addr) {
+        //Throws an exception if address is not valid
+        //And causes the hook to fail
+        ethers.utils.getAddress(addr);
+        return BigNumber.from(ethers.utils.parseEther('1.0'));
+      }
+      decimals() {
+        return ethers.BigNumber.from('18');
+      }
+    },
+  },
+}));
+
+describe('useTokenBalanceTest', () => {
+  test('happy path - returns accountsAddresses balance if both accountAddress and tokenAddress are defined  ', async () => {
+    const wrapper = ({ children }) => (
+      <>
+        {/* @ts-ignore */}
```

Comment:
I've added ts-ignore on purpose, so we can pass a mocked provider object. We need this to pass undefined checks in the actual hook. However, the provider is never used because we are mocking ethers.js so how to tell the compiler that it's fine to not pass an actual implementation of provider?